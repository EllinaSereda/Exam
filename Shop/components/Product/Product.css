.ProductComponent img{
    width:70px;

}
.ProductComponent  .big{
    display:block;
    width:400px;
    margin:10px auto;
    margin-bottom:30px;
}


.ProductComponent  h3{
    font-size:24px;
    margin:10px auto;
}
.ProductComponent  h2{
    font-size:30px;
    margin:10px auto;
    margin-bottom:40px;
}
.ProductComponent  .Variant{
    position: relative;
    width:80%;
    margin: auto;
    clear:both;
    text-align: left;
    
}
.ProductComponent  .Variant img{
    padding:20px;
    float:left;
    width:70px;
    height:100%;
}
.ProductComponent  .Variant h5{
    font-size: 22px;
    color:rgb(84, 84, 84);
    color:rgba(0, 0, 0, 0.3);
    margin-top:25px;
    margin-bottom:15px;
}
.ProductComponent  .Variant span{
    font-size: 18px;
    margin-top:25px;
    margin-bottom:15px;

}
.ProductComponent  .Variant{
    border-top:1px solid black;
}
.ProductComponent  .Variant:first-child{
    border-top:1px solid white;
}

.ProductComponent  .order{
    float:right;
    margin-top:50px;
    overflow: hidden;
}
.ProductComponent .order{
    float:right;
    margin:30px 0;
    overflow: hidden;
}
.ProductComponent  .order .Get, .ProductComponent  .Message {
    font-family: 'Ubuntu', sans-serif;
    color:white;
    border-radius:10px;
    font-size:18px;
    padding:10px 30px;
    float:right;
}

.ProductComponent  .order .Get{
    background-color:rgba(156, 60, 118, 0.7);
}
.ProductComponent  .Message{
    margin:35px 0;
    background-color:rgba(213, 195, 221, 0.7);
    
}
.ProductComponent .order .Get:hover{
    background-color:rgba(156, 60, 118, 1);
}
.ProductComponent  .Message:hover{
    background-color:#81597E;
}

.ProductComponent  .order .buttons{
    padding:0 10px 0 8px;
    width:108px;
    float:right;
    border-bottom:2px solid #9C3C76;
    margin:0 10px;
}
.ProductComponent  .order .buttons span{
    font-size:30px;
}

.ProductComponent  .order input[type="number"]{
    border:none;
    outline:none;
    width:70px;
    font-size:18px;
    margin:10px 5px 8px 5px;
    text-align: center;
} 

/* Убирает стрелочки */
.ProductComponent  .order input::-webkit-outer-spin-button,
.ProductComponent  .order input::-webkit-inner-spin-button {
    -webkit-appearance: none;
}

.ProductComponent  .order input[type='number'] {
    -moz-appearance: textfield;
}
.ProductComponent  .descr{
    margin:10px;
}

