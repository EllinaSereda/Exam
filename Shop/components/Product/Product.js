import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from '../../events';
import {connect} from 'react-redux';
import firebase from 'firebase';
import { NavLink } from 'react-router-dom';
import {product_addtobusket} from '../../redux/UserAC';
import './Product.css';
class Product extends React.PureComponent {

  static propTypes = {
    brands:PropTypes.shape(
      {
        brands:PropTypes.array,
        brandsMain:PropTypes.array,
      }),
    amount:PropTypes.object,
  };
  
  state={
    info:null,
    bigImg:0,
    content:0,
    n:0,
    amount:null,
  }
  
  componentDidMount(){ 
    let hash={};
    this.props.prod.stock.forEach(v => {
        hash[v.vol]=1;
      });
      this.setState({amount:hash})
  }
  
  setBigImg=(EO)=>{             
  this.setState({bigImg:EO.target.alt})
  }

  setContent=(EO)=>{
    this.setState({content:Number(EO.target.id)})
  }
  //  обработчик кнопки -
  dec=(EO)=>{   
    let newState={...this.state.amount};
    if(0<newState[parseInt(EO.target.id)]){
      newState[parseInt(EO.target.id)]-=1;
    }
    this.setState({amount:newState}); 
    //voteEvents.emit('SetAm',newState);
  }
 //  обработчик кнопки +
  inc=(EO)=>{
    let newState={...this.state.amount};
    let product=this.props.prod.stock.filter(v=>v.vol==parseInt(EO.target.id))[0];
    if(product.in>newState[parseInt(EO.target.id)]){
      newState[parseInt(EO.target.id)]+=1;
    }
    this.setState({amount:newState}); 
    //voteEvents.emit('SetAm',newState);
  }
 
 
  setAmount=(EO)=>{
    let newState={...this.state.amount};
    newState[EO.target.id]=Number(EO.target.value);
    this.setState({amount:newState});
   // voteEvents.emit('SetAm',newState);
  }
  addToBusket=(EO)=>{
    if(EO.target.id>0){
        EO.amount=this.state.amount[EO.target.id];
        voteEvents.emit('AddBusket',EO);
    }
  }

  render() {
    let img=[];
    img.push(this.props.prod.img.map((v,i)=><img alt={i} key={i} className='small' onClick={this.setBigImg} src={v}/>));
    return <div key={this.props.prod.code} className='ProductComponent'>
        <h3>{this.props.prod.brand}</h3>
        <h2>{this.props.prod.name}</h2>
        {this.props.prod.img.length>1 ?
          <div key="img">
            <img key={this.state.bigImg} className='big' src={this.props.prod.img.filter((v,i)=>i==this.state.bigImg)[0]}/>
            <div>{img}</div>
          </div> 
          :<img className='big' src={this.props.prod.img[0]}/>
        }                                                                                                         
        <div key="buy" className="buy">
        {         //Форма покупки товара
          this.props.prod.stock.map((x,i)=>{return <div key={i} className="Variant">
          {(x.in!=0)?
          <div className="order">
            <input type="button" className="Get" id={x.vol} value="В корзину" onClick={this.addToBusket}/>
            <div className="buttons">
              <span id={x.vol+"-"} onClick={this.dec}>-</span>
              <input type="number" value={this.state.amount?this.state.amount[x.vol]:"1"} min={1} id={x.vol} onChange={this.setAmount}/>
              <span id={x.vol+"+"} onClick={this.inc}>+</span>
            </div>

          </div>
          :<input type="button" className="Message" value="Сообщить о наличии"/>
          }
          <img src={this.props.prod.img[0]}/>
          <h5>{this.props.prod.brand}</h5>
          <div key="descr" className="descr">
            <span>{this.props.prod.name} {x.vol} мл </span>
            <span>{x.price} руб.</span>
          </div>
          </div>}) 
        }</div>
  </div>;
   
    

  }

}



export default Product;
