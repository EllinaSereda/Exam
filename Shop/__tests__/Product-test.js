"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import Product from '../components/Product/Product';

test('работа Product', () => {

  // создаём тестовую версию компонента
  const component = renderer.create(
    <Product img={["#","#"]} prod={{brand:"Carol",code:0,img:["#","#"],info:"parfume",name:"Good",sex:"w",stock:[{in:0,price:122,vol:30}]}}/>
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
});
