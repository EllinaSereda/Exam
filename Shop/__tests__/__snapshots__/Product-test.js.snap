// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Product 1`] = `
<div
  className="ProductComponent"
>
  <h3>
    Carol
  </h3>
  <h2>
    Good
  </h2>
  <div>
    <img
      className="big"
      src="#"
    />
    <div>
      <img
        alt={0}
        className="small"
        onClick={[Function]}
        src="#"
      />
      <img
        alt={1}
        className="small"
        onClick={[Function]}
        src="#"
      />
    </div>
  </div>
  <div
    className="buy"
  >
    <div
      className="Variant"
    >
      <input
        className="Message"
        type="button"
        value="Сообщить о наличии"
      />
      <img
        src="#"
      />
      <h5>
        Carol
      </h5>
      <div
        className="descr"
      >
        <span>
          Good
           
          30
           мл 
        </span>
        <span>
          122
           руб.
        </span>
      </div>
    </div>
  </div>
</div>
`;
